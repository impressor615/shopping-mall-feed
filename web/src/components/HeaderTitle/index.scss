@import "~@/assets/styles/variables";

.header-title {
  span:first-child {
    margin-right: 15px;
  }

  span:last-child {
    font-size: 15px;
    color: $secondary;
  }
}