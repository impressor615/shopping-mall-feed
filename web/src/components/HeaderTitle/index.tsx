import "./index.scss";

import React from "react";

interface HeaderTitleProps {
  title: string;
}

export const getToday = (): string => {
  const currentYear = new Date().getFullYear().toString().substr(-2);
  const currentMonth = new Date().getMonth() + 1;
  const currentDate = new Date().getDate();
  return `${currentYear}. ${currentMonth}. ${currentDate}`;
};

const HeaderTitle = ({ title }: HeaderTitleProps) => (
  <h3 className="header-title">
    <span>{title}</span>
    <span>{getToday()}</span>
  </h3>
);

export default HeaderTitle;
