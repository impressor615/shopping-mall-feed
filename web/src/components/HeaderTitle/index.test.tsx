import enzyme from "enzyme";
import React from "react";

import HeaderTitle, { getToday } from "@/components/Headertitle";

describe("HeaderTitle", () => {
  const title = "오늘 뭐 입지?";
  it("should rendered texts should be matched with prop title", () => {
    const wrapper = enzyme.shallow(<HeaderTitle title={title} />);
    expect(wrapper.find(".header-title span:first-child").text()).toEqual(title);
  });

  it("should render today's date", () => {
    const today = getToday();
    const wrapper = enzyme.shallow(<HeaderTitle title={title} />);
    expect(wrapper.find(".header-title span:last-child").text()).toEqual(today);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<HeaderTitle title={title} />);
    expect(wrapper).toMatchSnapshot();
  });
});
