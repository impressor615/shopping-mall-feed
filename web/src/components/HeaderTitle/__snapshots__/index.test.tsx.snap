// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle should be matched with snapshot 1`] = `
<h1
  className="header-title"
>
  <span>
    오늘 뭐 입지?
  </span>
  <span>
    19. 2. 9
  </span>
</h1>
`;
